dashboard/app/models/user.rb

```diff
@@ -126,25 +126,42 @@ class User < ActiveRecord::Base
   # You can be associated with districts through cohorts
   # has_many :districts, through: :cohorts.
 
+  # @param permission [UserPermission] the permission to query.
+  # @return [Boolean] whether the User has permission granted.
+  # TODO(asher): Determine whether request level caching is sufficient, or
+  #   whether a memcache or otherwise should be employed.
+  def permission_from_cache?(permission)
+    # Default to using the value cached in the instance variable, if present.
+    unless @permissions.nil?
```

Comment:
Can remove duplicate `return @permissions.include? permission` by removing `unless` block,
changing line below to `@permissions = UserPermission.where(user_id: id).pluck(:permission) unless !@permissions.nil?`